zerver/migrations/0469_remove_message_upper_subject_idx_and_more.py

```diff
@@ -0,0 +1,111 @@
+                models.F("recipient_id"),
+                name="zerver_message_realm_sender_recipient",
+            ),
+        ),
+        AddIndexConcurrently(
+            model_name="message",
+            index=models.Index(
+                models.F("realm_id"),
+                models.F("sending_client_id"),
+                name="zerver_message_realm_sending_client",
+            ),
+        ),
+        # upper(subject)
+        AddIndexConcurrently(
+            model_name="message",
+            index=models.Index(
+                models.F("realm_id"),
+                django.db.models.functions.text.Upper("subject"),
+                name="zerver_message_realm_upper_subject",
+            ),
+        ),
+        RemoveIndexConcurrently(
+            model_name="message",
+            name="upper_subject_idx",
+        ),
+        # date_sent
+        AddIndexConcurrently(
+            model_name="message",
+            index=models.Index(
+                models.F("realm_id"), models.F("date_sent"), name="zerver_message_realm_date_sent"
+            ),
+        ),
+        RemoveIndexConcurrently(
+            model_name="message",
+            name="zerver_message_date_sent_3b5b05d8",
+        ),
+        # recipient / upper(subject) / id
+        AddIndexConcurrently(
+            model_name="message",
+            index=models.Index(
+                models.F("realm_id"),
+                models.F("recipient_id"),
+                django.db.models.functions.text.Upper("subject"),
+                models.OrderBy(models.F("id"), descending=True, nulls_last=True),
+                name="zerver_message_realm_recipient_upper_subject",
+            ),
+        ),
+        RemoveIndexConcurrently(
+            model_name="message",
+            name="zerver_message_recipient_upper_subject",
+        ),
+        # recipient / subject / id
+        AddIndexConcurrently(
+            model_name="message",
+            index=models.Index(
+                models.F("realm_id"),
+                models.F("recipient_id"),
+                models.F("subject"),
+                models.OrderBy(models.F("id"), descending=True, nulls_last=True),
+                name="zerver_message_realm_recipient_subject",
+            ),
+        ),
+        RemoveIndexConcurrently(
+            model_name="message",
+            name="zerver_message_recipient_subject",
+        ),
+        # sender / delivered=False / delivery_type / scheduled_timestamp
+        AddIndexConcurrently(
+            model_name="scheduledmessage",
+            index=models.Index(
+                condition=models.Q(("delivered", False)),
+                fields=["realm_id", "sender", "delivery_type", "scheduled_timestamp"],
+                name="zerver_realm_unsent_scheduled_messages_by_user",
+            ),
+        ),
+        RemoveIndexConcurrently(
+            model_name="scheduledmessage",
+            name="zerver_unsent_scheduled_messages_by_user",
+        ),
+    ]
```

Comment:
Another option for this migration would be to just remove the old indexes in a follow-up commit.